test/libyul/yulOptimizerTests/fullSuite/loopInvariantCodeMotion.yul

```diff
@@ -29,7 +29,7 @@
 //         let length := calldataload(_1)
 //         let i := 0
 //         let _2 := calldataload(7)
-//         for { } true { i := add(i, 1) }
+//         for { } 1 { i := add(i, 1) }
```

Comment:
It doesn't really matter which of these we get here - but did you figure out why this happens? It's a bit weird, since the only use of the block hasher should be as pre-filtering in the equivalent function detector/combiner...

Reply:
it's due to the expression hasher (used in CSE). in the [devcon_example](https://github.com/ethereum/solidity/blob/develop/test/libyul/yulOptimizerTests/fullSuite/devcon_example.yul) the relevant steps are

```js
let i := 0
let i_17 := i
let _5 := 1
for { }
true
{ ... } { ... }
```

becomes

```js
let i := 0
let i_17 := i
let _5 := 1
for { } 
_5 
{ ... } { ... }
```

where the the `_5` is the extracted `1` from `{ i := add(i, 1) }`

Reply:
That would make sense - but I must be blind, I don't see the expression hasher in the CSE :-)... oh wow, now I see it... as ``ExpressionHash`` in the hash map there... alright, that explains it :-D...